Configuration fails when trying to compile HIPFORT with a CUDA backend

### Problem Description

I need to ensure that Hipfort works with an **nvidia** backend. However when I try to install **hipfort-6.0.2** the CMAKE configuration fails when I set **HIP_PLATFORM="nvidia"**. 

I have an RTX 3060 and CUDA 12.4 installed via package manager and everything seems ok with the NVIDIA software stack. 

### Operating System

Ubuntu 22.04.4 LTS (Jammy Jellyfish)

### CPU

AMD Ryzen 7 6800H

### GPU

RTX 3060

### ROCm Version

ROCm 6.0.2

### ROCm Component

hipfort

### Steps to Reproduce

1. Install NVIDIA driver (Driver 550 installed)
2. Install rocm 6.0.2 using amdgpu-install
4. Install the **hip-runtime-nvidia** package (CUDA 12.4 installed, and I have an RTX 3060 with working drivers)
5. Uninstall **hipfort-dev** using the package manager
6. Download and unpack **hipfort-rocm-6.0.2**
7. Change directory to unpacked sources
8. Make a **build** directory

Run the following script from the **build** directory

```bash
FLAGS="-O2 -g -free -cpp -fPIC -fmax-errors=5 -fno-underscoring"

export HIP_PLATFORM="nvidia"

CFLAGS="--offload-arch=sm_86 -lcublas -lcusolver -lcufft"

cmake -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR -DHIPFORT_COMPILER=$(which gfortran) -DCMAKE_Fortran_FLAGS="${FLAGS}" -DCMAKE_C_COMPILER=$(which gcc) -DCMAKE_CXX_COMPILER=$(which g++) -DCMAKE_Fortran_COMPILER=$(which gfortran)  .. 
```

I then get numerous configuration errors to say that another target already exists.

```
CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocblas/rocblas-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:126 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocblas/rocblas-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:126 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocblas/rocblas-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:126 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipblas/hipblas-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:133 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipblas/hipblas-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:133 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipblas/hipblas-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:133 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocfft/rocfft-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:140 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocfft/rocfft-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:140 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocfft/rocfft-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:140 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipfft/hipfft-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:147 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipfft/hipfft-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:147 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipfft/hipfft-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:147 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocrand/rocrand-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:154 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocrand/rocrand-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:154 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocrand/rocrand-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:154 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hiprand/hiprand-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:161 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hiprand/hiprand-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:161 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hiprand/hiprand-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:161 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocsolver/rocsolver-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:168 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocsolver/rocsolver-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:168 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocsolver/rocsolver-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:168 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocblas/rocblas-config.cmake:90 (find_dependency)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocsolver/rocsolver-config.cmake:92 (find_dependency)
  lib/CMakeLists.txt:168 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocblas/rocblas-config.cmake:90 (find_dependency)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocsolver/rocsolver-config.cmake:92 (find_dependency)
  lib/CMakeLists.txt:168 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocblas/rocblas-config.cmake:90 (find_dependency)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocsolver/rocsolver-config.cmake:92 (find_dependency)
  lib/CMakeLists.txt:168 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipsolver/hipsolver-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:175 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipsolver/hipsolver-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:175 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipsolver/hipsolver-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:175 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocsparse/rocsparse-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:182 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocsparse/rocsparse-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:182 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/rocsparse/rocsparse-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:182 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:21 (add_library):
  add_library cannot create imported target "hip::device" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipsparse/hipsparse-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:189 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:22 (add_library):
  add_library cannot create imported target "hip::host" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipsparse/hipsparse-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:189 (find_package)


CMake Error at /opt/rocm-6.0.2/lib/cmake/hip/hip-config-nvidia.cmake:23 (add_library):
  add_library cannot create imported target "hip::amdhip64" because another
  target with the same name already exists.
Call Stack (most recent call first):
  /opt/rocm-6.0.2/lib/cmake/hip/hip-config.cmake:153 (include)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm-6.0.2/lib/cmake/hipsparse/hipsparse-config.cmake:90 (find_dependency)
  lib/CMakeLists.txt:189 (find_package)
```

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information

Environment variables

HIP_PLATFORM="nvidia"

--- a/docs/install/install.rst
+++ b/docs/install/install.rst
@@ -27,8 +27,22 @@ Building and testing hipFORT from source
       make -C build check
 
 .. note::
-        You can control the location of the hipFORT installation by setting the ``HIPFORT_INSTALL_DIR`` build variable.
         The hipFORT installation will compile backends for both ROCm (``hipfort-amdgcn``) and CUDA (``hipfort-nvptx``) backends. When installing hipFORT from source, you do not need to specify the `HIP_PLATFORM` environment variable.
+
+Customizing the build
+=======================
+You can customize the build by setting the following environment variables:
+
+* `HIPFORT_COMPILER` - Fortran compiler to be used
+* `HIPFORT_AR` - Static archive command
+* `HIPFORT_RANLIB` - ranlib used to create Static archive
+* `HIPFORT_COMPILER_FLAGS` - Compiler flags to build hipFORT
+* `HIPFORT_BUILD_TYPE` - Set to RELEASE TESTING or DEBUG
+* `HIPFORT_INSTALL_DIR` - Install directory for hipFORT
+
+.. note::
+        Setting the `CMAKE_INSTALL_DIR` build variable will **not** influence the installation location.
+        Setting the `CMAKE_Fortran_COMPILER` build variable will **not** influence the Fortran compiler used to build hipFORT.
 
 Fortran interfaces
 ===================

--- a/docs/install/install.rst
+++ b/docs/install/install.rst
@@ -26,6 +26,10 @@ Building and testing hipFORT from source
       make -C build
       make -C build check
 
+.. note::
+        You can control the location of the hipFORT installation by setting the ``HIPFORT_INSTALL_DIR`` build variable.
+        The hipFORT installation will compile backends for both ROCm (``hipfort-amdgcn``) and CUDA (``hipfort-nvptx``) backends. When installing hipFORT from source, you do not need to specify the `HIP_PLATFORM` environment variable.
+
 Fortran interfaces
 ===================
 

